### PR TITLE
Added polygon plotter

### PR DESCRIPTION
**Pull request by ctessum**
PTAL

### CODE REVIEW

File: plotter/polygon_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+// ExamplePolygon_hexagons creates a heat map with hexagon shapes.
+// The output of this example is at
+// https://github.com/gonum/plot/blob/master/plotter/testdata/polygon_hexagons_golden.png.
+func ExamplePolygon_hexagons() {
+	// hex returns a hexagon centered at (x,y) with radius r.
+	hex := func(x, y, r float64) XYs {
+		g := make(XYs, 6)
+		for i := 0; i < 6; i++ {
+			g[i].X = x + r*math.Cos(math.Pi*2/6*float64(i))
```
Comment (vladimir-ch) [inline plotter/polygon_test.go]: `math.Pi/3*float64(i)`

File: plotter/polygon.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+type Polygon struct {
+	// XYs is a copy of the vertices of this polygon.
+	XYs []XYs
+
+	// LineStyle is the style of the line around the edge
+	// of the polygon.
+	draw.LineStyle
+
+	// Color is the fill color of the polygon.
+	Color color.Color
```
Comment (ctessum) [inline plotter/polygon.go]: This is potentially different for different backends. One of the tests documents the behavior for the img backend. I can add some text to clarify this and also tests for the other default backends.

File: plotter/polygon.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import (
+	"image/color"
+	"math"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// Polygon implements the Plotter interface, drawing a polygon.
```
Comment (ctessum) [inline plotter/polygon.go]: I added this text to the NewPolygon documentation.

File: plotter/polygon.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+type Polygon struct {
+	// XYs is a copy of the vertices of this polygon.
```
Comment (ctessum) [inline plotter/polygon.go]: Done

**Comment by ctessum** [comment]
The different backends treat holes differently by default, and possibly self intersections as well but I haven't checked that. (Self-intersections are often not allowed in geometry specifications so I'm less concerned about them than the hole behavior.) My feeling is that fixing this is an issue for a separate pull request, possibly involving adding the ability to specify fill-rules to the vg.Canvas interface, but let me know if anyone feels otherwise.

**Comment by sbinet** [comment]
the travis failure is real. @ctessum please rebase on top of the latest `master` :)
@vladimir-ch are you OK with the latest PR?

**Comment by vladimir-ch** [comment]
I'll take a look today.

File: plotter/polygon_test.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+	}
+	err = p.Save(100, 100, "testdata/polygon_holes.pdf")
+	if err != nil {
+		log.Panic(err)
+	}
+	err = p.Save(100, 100, "testdata/polygon_holes.eps")
+	if err != nil {
+		log.Panic(err)
+	}
+
```
Comment (vladimir-ch) [inline plotter/polygon_test.go]: Remove empty line.

File: plotter/polygon_test.go [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+	const (
+		r = math.Pi / 4 // r is the hexagon radius
+		// x0 and y0 are the beginning coordinates for the hexagon plot
+		x0 = 0.0
+		y0 = 0.0
+		// nx and ny are the number of hexagons in the x and y directions.
+		nx = 5
+		ny = 5
+	)
+	// dx and dy are the distance between hexgons
```
Comment (vladimir-ch) [inline plotter/polygon_test.go]: hexagons.

File: plotter/polygon_test.go [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+	p.X.Label.Text = "X"
+	p.Y.Label.Text = "Y"
+
+	colorMap := moreland.SmoothBlueRed()
+	colorMap.SetMax(2)
+	colorMap.SetMin(-2)
+	colorMap.SetConvergePoint(0)
+
+	const (
+		r = math.Pi / 4 // r is the hexagon radius
```
Comment (vladimir-ch) [inline plotter/polygon_test.go]: Dot at the end here and the next line.

File: plotter/polygon.go [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+		ymini, ymaxi := Range(YValues{ring})
+		xmin = math.Min(xmin, xmini)
+		xmax = math.Max(xmax, xmaxi)
+		ymin = math.Min(ymin, ymini)
+		ymax = math.Max(ymax, ymaxi)
+	}
+	return
+}
+
+// Thumbnail the thumbnail for the Polygon,
```
Comment (kortschak) [inline plotter/polygon.go]: Interestingly, [it's not](https://bellsiebooks.files.wordpress.com/2013/08/verbing_weirds_language.gif), but adding "creates" in the appropriate place would be nicer.

**Pull request merged**